packages/code-editor/src/CodeEditor/CodeEditor.tsx

```diff
@@ -36,35 +51,76 @@ const defaultCodeEditorOptions: EditorProps["options"] = {
   },
 };
 
+/** Languages we have custom validations for. */
+const languages = { xml: "xml" };
+const completionProviders: Record<
+  string,
+  (monaco: Monaco, elements?: Elements) => object
+> = {
```

Comment:
Looking at how this code is structured (all registering the `xml` key), it looks like a first-stage of a plugin system. We could evolve 

```tsx
const languagePlugins = [
  xmlLanguagePlugin(),
  // returns { completionProviders, validationMarkers, etc. }
  // or ["xml", { completionProviders, validationMarkers, etc. }]
]
```

Or maybe an object will be better for now (Array would be more versatile, but needlessly more complex?):

```tsx
const languagePlugins = {
  xml: xmlLanguagePlugin(), // returns completionProviders, validationMarkers, etc.
}
```

Reply:
Makes sense 🙏 I'll update the code to follow the object approach for now.